### ISSUE TITLE
High  % of reads unmapped: too short |       97.77% 

### ISSUE THREAD

**Issue by lovelymaoqin** [alexdobin/STAR#1046]
I am new user of STAR, and I am mapping my miRNA fastq file, and my data is 50bp, HS4000 single end run! but the number of reads unmapped and flagged as short is over 97%. Not sure why, any suggestion would be appreciated! This is my log files:

     Started job on |       Sep 26 15:33:52
                             Started mapping on |       Sep 26 15:34:41
                                    Finished on |       Sep 26 17:16:02
       Mapping speed, Million of reads per hour |       29.32
                          Number of input reads |       49522582
                      Average input read length |       50
                                    UNIQUE READS:
                   Uniquely mapped reads number |       268105
                        Uniquely mapped reads % |       0.54%
                          Average mapped length |       37.26
                       Number of splices: Total |       18640
            Number of splices: Annotated (sjdb) |       112
                       Number of splices: GT/AG |       12929
                       Number of splices: GC/AG |       92
                       Number of splices: AT/AC |       4
               Number of splices: Non-canonical |       5615
                      Mismatch rate per base, % |       1.49%
                         Deletion rate per base |       0.00%
                        Deletion average length |       1.48
                        Insertion rate per base |       0.00%
                       Insertion average length |       1.00
                             MULTI-MAPPING READS:
        Number of reads mapped to multiple loci |       714745
             % of reads mapped to multiple loci |       1.44%
        Number of reads mapped to too many loci |       123058
             % of reads mapped to too many loci |       0.25%
                                  UNMAPPED READS:
  Number of reads unmapped: too many mismatches |       0
       % of reads unmapped: too many mismatches |       0.00%
            Number of reads unmapped: too short |       48416512
                 % of reads unmapped: too short |       97.77%
                Number of reads unmapped: other |       162
                     % of reads unmapped: other |       0.00%
                                  CHIMERIC READS:
                       Number of chimeric reads |       0
                            % of chimeric reads |       0.00%
[maoqin20@longleaf-login2 startest]$ tail Log.final.out
                                  UNMAPPED READS:
  Number of reads unmapped: too many mismatches |       0
       % of reads unmapped: too many mismatches |       0.00%
            Number of reads unmapped: too short |       48416512
                 % of reads unmapped: too short |       97.77%
                Number of reads unmapped: other |       162
                     % of reads unmapped: other |       0.00%
                                  CHIMERIC READS:
                       Number of chimeric reads |       0
                            % of chimeric reads |       0.00%

**Comment by alexdobin** [comment]
Hi @lovelymaoqin 

miRNA reads will be mostly ~22b, so you have to trim the adapters at 3'.
You can use an external trimmer (e.g. cutadapt) or internal STAR trimmer with
`--clip3pAdapterSeq <3'AdapterSeq>`

Cheers
Alex

**Comment by lovelymaoqin** [comment]
Hi Alex,

Thanks for your suggestion. I used cutadapt totrim the adapter, and this is my Log.final.out. The unique mapped read is 69.56%, is not high. Is that normal? 
![image](https://user-images.githubusercontent.com/62307133/95146893-2506a200-074d-11eb-9ba5-8aadf772c29a.png)

**Comment by scseekers** [comment]
Yes around 70-90 percent of uniquely mapping reads are considered good. At least in my lab. Also, it depends upon what kind of genome you are aligning to. For instance in rice, getting 35-50% unique reads is also [fine.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2928502/)

**Comment by lovelymaoqin** [comment]
> Yes around 70-90 percent of uniquely mapping reads are considered good. At least in my lab. Also, it depends upon what kind of genome you are aligning to. For instance in rice, getting 35-50% unique reads is also [fine.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2928502/)

Thanks. My genome is human. One more question is about the average input read length, does this depend on the library?

**Comment by RochaLAJ** [comment]
Yep, it depends of your library and protocol of sequencing. Here i work with whole blood RNA-Seq Transcriptome and the average fragment size is 90~120bp for most transcripts detected.

**Comment by lovelymaoqin** [comment]
> Yep, it depends of your library and protocol of sequencing. Here i work with whole blood RNA-Seq Transcriptome and the average fragment size is 90~120bp for most transcripts detected.

Thanks so much.

**Comment by alexdobin** [comment]
Hi @lovelymaoqin 

the Log.final.out says:
Average input read length |       50
If your input reads were 50b, it seems to me that the trimming did not work.

Cheers
Alex

**Comment by lovelymaoqin** [comment]
> Hi @lovelymaoqin
> 
> the Log.final.out says:
> Average input read length | 50
> If your input reads were 50b, it seems to me that the trimming did not work.
> 
> Cheers
> Alex

Hi Alex,

I reran it after trimming, this is the output:
![image](https://user-images.githubusercontent.com/62307133/96189891-592d5000-0f0f-11eb-9a37-7b2ae5d5573e.png)

And one more question abut the unique mapping rate, any threshold for human? Does 68% look good?

Thanks.

**Comment by alexdobin** [comment]
Hi @lovelymaoqin 

68% is good for small RNA sequencing. You will probably find that the multimappers are fragments of longer non-coding RNAs (tRNA, snoRNA,rRNA etc).

Cheers
Alex
